balancer-js/src/modules/pools/pool-types/concerns/weighted/priceImpact.concern.ts

```diff
@@ -0,0 +1,61 @@
+import { cloneDeep } from 'lodash';
+import { SubgraphPoolBase, WeightedPool } from '@balancer-labs/sor';
+import { PriceImpactConcern } from '@/modules/pools/pool-types/concerns/types';
+import { parseToBigInt18 } from '@/lib/utils/math';
+import { calcPriceImpact } from '@/modules/pricing/priceImpact';
+import {
+  ONE,
+  BZERO,
+  _computeScalingFactor,
+  _upscale,
+} from '@/lib/utils/solidityMaths';
+import { BalancerError, BalancerErrorCode } from '@/balancerErrors';
+import { Pool } from '@/types';
+
+export class WeightedPoolPriceImpact implements PriceImpactConcern {
+  /**
+   * Calculates the BPT return amount when investing with no price impact.
+   * @param { Pool } pool Investment pool.
+   * @param { bigint [] } amounts Token amounts being invested. EVM Scale. Needs a value for each pool token.
+   * @returns { bigint } BPT amount.
+   */
+  bptZeroPriceImpact(pool: Pool, tokenAmounts: bigint[]): bigint {
+    if (tokenAmounts.length !== pool.tokensList.length)
+      throw new BalancerError(BalancerErrorCode.INPUT_LENGTH_MISMATCH);
+
+    // swapFee, totalShares, totalWeight all scaled up to 18 decimals
+    const weightedPool = WeightedPool.fromPool(pool as SubgraphPoolBase);
```

Comment:
Same here about being able to use Pool and parsePoolInfo instead of SubgraphPoolBase.